### ISSUE TITLE
New default.link file breaks previously working udevadm command

### ISSUE THREAD

**Issue by bitfehler** [systemd/systemd#13035]
**systemd version the issue has been seen with**
> systemd 242 (242.32-3-arch)

**Used distribution**
 > Arch Linux

**Expected behaviour you didn't see**
 > `udevadm test-builtin net_setup_link /sys/class/net/wlp2s0` runs successfully

**Unexpected behaviour you saw**
 > `udevadm test-builtin net_setup_link /sys/class/net/wlp2s0` fails because it ignores the default .link file, output:

```
# udevadm -d test-builtin net_setup_link /sys/class/net/wlp2s0 
Trying to open "/etc/systemd/hwdb/hwdb.bin"...
Trying to open "/etc/udev/hwdb.bin"...
Trying to open "/usr/lib/systemd/hwdb/hwdb.bin"...
Trying to open "/usr/lib/udev/hwdb.bin"...
=== trie on-disk ===
tool version:          242
file size:         9969543 bytes
header size             80 bytes
strings            2090215 bytes
nodes              7879248 bytes
Load module index
Found container virtualization none.
timestamp of '/etc/systemd/network' changed
timestamp of '/usr/lib/systemd/network' changed
Parsed configuration file /usr/lib/systemd/network/99-default.link
Created link configuration context.
ID_NET_DRIVER=ath10k_pci
wlp2s0: Device has name_assign_type=4
Config file /usr/lib/systemd/network/99-default.link matches device based on renamed interface name 'wlp2s0', ignoring
wlp2s0: No matching link configuration found.
Builtin command 'net_setup_link' fails: No such file or directory
Unload module index
Unloaded link configuration context.
(command failed)
```

**Steps to reproduce the problem**
 * Upgrade to systemd version that includes this change: https://github.com/systemd/systemd/commit/84ea567eb4326eb970a33188649fde6bea2a0d4e#diff-fd460a87ba85f4b05ee20317be02b21aR11
 * run `udevadm test-builtin net_setup_link` on some interface that is not covered by any specific .link file

This change in behavior breaks netplan, which calls the above command. I am not quite sure if there are any reservations about calling the command in the first place, but it did work while the match section was still empty, so I do think this could be considered a regression?

And also, the warning added in the above-linked commit about an empty Match section says:
> The file will match all interfaces. If that is intended, please add OriginalName=* in the [Match] section.

This seems to not be quite true, because the file is now ignored in certain circumstances.

**Comment by yuwata** [comment]
Thank you for the report. Fix is waiting in #13044. If possible, please test the PR.

**Comment by bitfehler** [comment]
I can confirm that #13044 fixes the issue for me. Impressive response time. Thanks a bunch!
